python/ray/train/base_trainer.py

```diff
@@ -351,25 +410,41 @@ def fit(self) -> Result:
             TrainingFailedError: If any failures during the execution of
             ``self.as_trainable()``.
         """
-        from ray.tune.tuner import Tuner
-        from ray.tune.error import TuneError
+        from ray.tune.tuner import Tuner, TunerInternal
 
         trainable = self.as_trainable()
         param_space = self._extract_fields_for_tuner_param_space()
 
-        tuner = Tuner(
-            trainable=trainable, param_space=param_space, run_config=self.run_config
-        )
+        if self._restore_path:
+            tuner = Tuner.restore(
+                self._restore_path,
+                overwrite_trainable=trainable,
+                resume_unfinished=True,
+                resume_errored=True,
```

Comment:
why set this to True? machine goes away will cause all the error trials to be retried is a bit weird?

Reply:
There is only ever 1 trial in the Trainer case. Do you think that this is not desired behavior on restore?

I'm thinking about the case where: Trainer errors due to some transient error, causing the script to fail. Job retries, and we want to resume the errored run.

Reply:
I agree with Justin, I think `resume_errored=True` is desired here